primitives/src/assets/currencies.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023-2024 Forecasting Technologies LTD.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+/// The `CurrencyClass` enum represents all non-ztg CurrencyClass
+// used in orml-tokens
```

Comment:
CurrencyClass represents all non-ZTG currencies. Orml-tokens handles non-volatile assets, that's why it's handling currencies. I'll keep the name `CurrencyClass`, because the asset class names describe what assets they hold, not which implementation handles them.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1181/commits/64058bab89c91960bc668f8a41c399481a8c68a1